src/server/index.js

```diff
@@ -1,22 +1,123 @@
+                    sentimentEntities: result.sentimented_entity_list.length
+                });
+            } else {
+                res.send(errorResponse.create(`Something went wrong. Status code: ${result.status.code}`));
+            }
+        } catch (e) {
+            console.log('Could not fetch sentiment result. Error:');
+            console.log(e);
+            res.send(errorResponse.create(`Unexpected error occurred. See logs for more details.`));
+        }
+    } else {
+        res.send(errorResponse.create(`Could not parse params`));
+    }
+});
+
+app.post('/summarize', async function (req, res) {
+    console.log(`URL param: ${req.body.url}`);
+    res.type('json');
+    const analysisResult = parsingResult.create("unknown", "");
+
+    if (req.body.url && req.body.url.trim() !== '') {
+        // Start the analysis process.
+        const responseLang = await fetch(`${api_base_url}lang-4.0/identification?key=${api_key}&url=${req.body.url}`, {method: "POST"});
+        try {
+            // Try to analyse the language of the given link.
+            const result = await responseLang.json();
+            if (result && result.language_list && result.language_list.length > 0) {
+                analysisResult.language = result.language_list[0].name;
+            }
+
+            // Prepare the summarization of the article.
+            const sentences = req.body.sentences ? req.body.sentences : 5;
+            const fetchUrl = `${api_base_url}summarization-1.0?key=${api_key}&of=json&sentences=${sentences}&url=${req.body.url}`;
+            console.log(`Trying to fetch with url: ${fetchUrl}`);
 
-app.get('/test', function (req, res) {
-    res.send(mockAPIResponse)
-})
+            const response = await fetch(fetchUrl);
+            try {
+                // Combine the results of the language detection and the summarization.
+                const result = await response.json();
+                if (result.status.code === "0") {
+                    console.log(`Credits left: ${result.status.remaining_credits}`);
+                    console.log(`Credits used: ${result.status.credits}`);
+                    analysisResult.summary = result.summary;
+                    res.send(analysisResult);
+                } else {
+                    res.send(errorResponse.create(`Something went wrong. Status code: ${result.status.code}`));
+                }
+            } catch (e) {
+                console.log('Could not fetch data. Error:');
+                console.log(e);
+                res.send(errorResponse.create(`Unexpected error occurred. See logs for more details.`));
+            }
+        } catch (e) {
+            console.log('Could not fetch language detection result. Error:');
+            console.log(e);
+            res.send(errorResponse.create(`Unexpected error occurred. See logs for more details.`));
+        }
+    } else {
+        res.send(errorResponse.create(`Could not parse params`));
+    }
+});
```

Comment:
This class needs to be refactored. It's too big and clunky. Not now, but if you would work in a team this would be an issue enough to reject your PR.